Sources/Utility/ArgumentParser.swift

```diff
@@ -447,3 +447,166 @@ public final class ArgumentParser: ArgumentParserProtocol {
         }
     }
 }
+
+/// A protocol representing an option structure, which can be
+/// bounded in the BoundParser.
+public protocol BoundParserOption {
```

Comment:
I would skip this protocol, and instead have `parse` just take the T to use. This allows a little more flexible use (you can even use the parser to parse multiple objects of the same type).